playlist.c

```diff
@@ -1545,21 +1610,42 @@ void playlist_write_file(playlist_t *playlist)
          json_write_space(context.writer, 4);
          JSON_Writer_WriteStartObject(context.writer);
 
-         json_write_new_line(context.writer);
-         json_write_space(context.writer, 6);
-         JSON_Writer_WriteString(context.writer, "path",
+         if (string_is_empty(playlist->entries[i].relative_path))
```

Comment:
You can make this a little more efficient by swapping the order of the two conditions and checking `if (playlist->entries[i].relative_path)`